### PR TITLE
Added e2e tests for page.added webhook

### PR DESCRIPTION
**Pull request by JohnGrisham**
Added e2e tests for page.added webhook

Ref - https://github.com/TryGhost/Ghost/issues/15537

### CODE REVIEW

File: ghost/core/test/e2e-webhooks/pages.test.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    stringMatching,
+    anyUuid,
+    anyLocalURL,
+    anyISODateTime,
+    anyObjectId,
+    anyContentVersion,
+    anyNumber
+} = matchers;
+
+const anyGhostAgent = stringMatching(
```
Comment (naz) [inline ghost/core/test/e2e-webhooks/pages.test.js:17]: The `anyGhostAgent` matcher is not a part of "matcher", so there's no need to re-declare it here 🙃  If you rebase your changes over the latest `main` the matcher should be available 👍

File: ghost/core/test/e2e-webhooks/pages.test.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+const buildAuthorSnapshot = (roles = false) => {
+    const authorSnapshot = {
+        last_seen: anyISODateTime,
+        created_at: anyISODateTime,
+        updated_at: anyISODateTime
+    };
+
+    // NOTE: this is such a bad hack! for the reasons I did not investigate the "add" event does not include
+    //       the roles but the "published" does! massive inconsistency and needs to be fixed one day
+    if (roles) {
```
Comment (naz) [inline ghost/core/test/e2e-webhooks/pages.test.js:36]: The 'roles' here does not seem to ever be "true", so probably no need for this whole block here

**Review by naz — CHANGES_REQUESTED** [review]
This is great! Thanks @JohnGrisham . I've requested couple very minor changes, otherwise it's ready to land in main 👍

File: ghost/core/test/e2e-webhooks/pages.test.js (hunk 1 above)
Comment (JohnGrisham) [reply ghost/core/test/e2e-webhooks/pages.test.js:17]: Thanks!

**Review by naz — APPROVED** [review]
The changes are good to go into main. Thanks @JohnGrisham 🙌  
A little note, could you please make sure the changes come squashed (as a single commit) and the commit message follows Ghost's [contributor guidelines](https://github.com/TryGhost/Ghost/blob/main/.github/CONTRIBUTING.md#commit-messages) 🙏

**Comment by JohnGrisham** [comment]
Changes were made and the commits consolidated into one. Thanks for your help!

**Comment by ErisDS** [comment]
Hi @JohnGrisham thanks so much for this PR. It has now been merged 🎉 

I'm not sure if you found this through hacktoberfest, but I've added the accepted label to this PR to make sure it counts. We have plenty of open issues and ongoing projects for hacktoberfest and beyond. We would love to see you around again 👋

**Pull request merged**